src/Microsoft.DotNet.CodeFormatting/Analyzers/ExplicitVariableTypeFixer.cs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SemanticModel model = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            TypeSyntax oldTypeSyntaxNode = null;
+            TypeSyntax newTypeSyntaxNode = null;
+            switch (GetRuleType(diagnostic))
+            {
+                case RuleType.RuleVariableDeclaration:
+                    var variableDeclarationNode = root.
+                        FindNode(diagnosticSpan).
+                        FirstAncestorOrSelf<VariableDeclarationSyntax>();
+                    Debug.Assert(variableDeclarationNode != null);
+                    oldTypeSyntaxNode = variableDeclarationNode.Type;
+                    // Implicit typed variables cannot have multiple declartors
+                    newTypeSyntaxNode = GetTypeSyntaxNode(variableDeclarationNode.Variables.Single(), model, cancellationToken);
+                    break;
+
+                case RuleType.RuleForEachStatement:
+                    var forEachStatementNode = root.
+                        FindToken(diagnosticSpan.Start).
+                        Parent.
+                        FirstAncestorOrSelf<ForEachStatementSyntax>();
+                    Debug.Assert(forEachStatementNode != null);
+                    oldTypeSyntaxNode = forEachStatementNode.Type;
+                    newTypeSyntaxNode = GetTypeSyntaxNode(forEachStatementNode, model, cancellationToken);
+                    break;
+
+                case RuleType.None:
+                    break;
+            }
+
+            if (oldTypeSyntaxNode != null && newTypeSyntaxNode != null)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        Resources.ExplicitVariableTypeFixer_Title,
+                        c => ReplaceVarWithExplicitType(context.Document, root, oldTypeSyntaxNode, newTypeSyntaxNode.WithTriviaFrom(oldTypeSyntaxNode))),
+                    diagnostic);
+            }
+        }
+
+        private Task<Document> ReplaceVarWithExplicitType(Document document, SyntaxNode root, SyntaxNode varNode, SyntaxNode ExplicitTypeNode)
+        {
+            return Task.FromResult(
+                document.WithSyntaxRoot(root.ReplaceNode(varNode, ExplicitTypeNode.WithAdditionalAnnotations(Simplifier.Annotation))));
+        }
+
+        // return null if could not construct a TypeSyntax with explicit type.
+        private static TypeSyntax GetTypeSyntaxNode(SyntaxNode node, SemanticModel model, CancellationToken cancellationToken)
```

Comment:
I'd call this `Make...` or `Create...` because to me "`Get`" implies that the thing already exists.